### DISCUSSION TITLE
Vips is not reading all tags (metadata)

### DISCUSSION THREAD

**Discussion by boan-jfm** [libvips/php-vips#219] [Q&A] [answered]
Hi

I used `exiftool` to write the 'Creator' tag (`exiftool -Creator="Bo Andersen" php-vips-meta-tags.jpg`) to an image, but when I try to read it using php-vips `getFields` or `vipsheader -a`, it's not included in the output.

![image](https://github.com/libvips/php-vips/assets/13201214/d6b7fd95-a76b-4622-b650-67243ab702f1)
_exiftool output - has the creator tag_

![image](https://github.com/libvips/php-vips/assets/13201214/f6d003e4-087c-47bf-888d-1e2b2a5e6209)
_vips output - missing the creator tag_ 

![php-vips-meta-tags](https://github.com/libvips/php-vips/assets/13201214/f4bff98a-0b17-43a1-a37c-1fa8d7d68211)
_Test image_

How do I get the information in php-vips?

Thanks.

**Comment by jcupitt** [comment, marked as answer]
Hi @boan-jfm,

`Creator` is part of XMP (I think), so libvips doesn't handle it for you directly. You'll need to get the `xmp-data` field and decode that using an xmp library.

You can see all the exif tags here:

https://exiftool.org/TagNames/EXIF.html

JPEGs also have ICC profiles (containing many internal metadata fields) and IPTC data blocks (with even more metadata).

**Reply by boan-jfm** [reply]
Thank you.

>Creator is part of XMP (I think), so libvips doesn't handle it for you directly.

Just out of curiosity, what types of metadata formats does libvips handle directly, is it "only" exif-tags?

**Reply by jcupitt** [reply]
It'll read and write all JPEG metadata, but it only decodes EXIF into separate tags.

You can't really read JPEG correctly without looking inside the exif, unfortunately, since you need that to handle things like camera orientation. But the other metadata can just be treated as byte arrays (from libvips point of view).

I suppose it reads inside ICC profiles too, but none of that internal detail is exposed in libvips metadata.
